File: 868 - Binary Gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        int left = -1, right = -1, gap = 0, i = 0;
        while (n > 0) {
            if (n % 2 == 1) {
                if (right == -1) {
                    right = i;
                } else {
                    left = i;
                    gap = gap > (left - right) ? gap : left - right;
                    right = left;
                }
            }
            i++;
            n /= 2;
        }
        return gap;
    }
};

File: 868 - Binary Gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        b = bin(n)[2:]
        left, right = -1, -1
        gap = 0
        for i, d in enumerate(b):
            if d == '1':
                if left == -1:
                    left = i
                else:
                    if right == -1:
                        right = i
                    gap = max(gap, right - left)
                    left = right
                    right = -1
        return gap